Toms-Storage-120/src/shared/java/com/tom/storagemod/gui/AbstractStorageTerminalScreen.java

```diff
@@ -65,6 +61,12 @@ public List<String> load(StoredItemStack key) throws Exception {
 		}
 
 	});
+	private static final LoadingCache<StoredItemStack, String> nbtCache = CacheBuilder.newBuilder().expireAfterAccess(5, TimeUnit.SECONDS).build(CacheLoader.from(
+			key -> String.valueOf(key.getStack().getTag())
+	));
+	private static final LoadingCache<StoredItemStack, List<String>> tagCache = CacheBuilder.newBuilder().expireAfterAccess(5, TimeUnit.SECONDS).build(CacheLoader.from(
+			key -> key.getStack().getTags().map(Object::toString).toList()
+	));
```

Comment:
Tag lookup is rather fast, but if you decide to cache it, there is no reason to cache the list just to toString it later.